decision_service/dsclientlib/ds_event.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+namespace decision_service {
+
+	//serializable ranking event
+	class ranking_event {
+	public:
+		ranking_event(const char* uuid, const char* context, const std::vector<std::pair<int, float>>& ranking, const std::string& model_id);
+		std::string serialize();
+
+	private:
+		const char* _uuid;
+		const char* _context;
+		const std::vector<std::pair<int, float>>& _ranking;
+		const std::string& _model_id;
```

Comment:
I'm worried about the life time of that reference

Reply:
switch to plain function instead of class

---
In reply to: [187049450](https://github.com/JohnLangford/vowpal_wabbit/pull/1479#discussion_r187049450) [](ancestors = 187049450)